test/client/components/campaign/joined_campaigns.test.js

```diff
@@ -29,14 +29,36 @@ describe('<JoinedCampaigns />', () => {
     it('should render 1 <SidebarList>', () => {
       expect(wrapper.find('SidebarList').length).toBe(1);
     });
-    it('should render 1 <CurrentCampaign>', () => {
-      expect(wrapper.find('CurrentCampaign').length).toBe(1);
-    });
   });
   describe('Mounting', () => {
     it('Should call the fetchQuestion action on mount', () => {
       mount(<JoinedCampaigns {...props} />);
       expect(props.fetchCampaignsByUser).toHaveBeenCalled();
     });
   });
+  describe('<CurrentCampaign> render switch', () => {
+    it('should render 1 <CurrentCampaign> when elements are present in current_campaign prop', () => {
+      const campaignsPresentProps = {
+        history: {
+          goBack: jest.fn(),
+          location: { pathname: ''}
+        },
+        auth: { id: 1 },
+        fetchCampaignsByUser: jest.fn(),
+        current_campaign: {},
+        joined_campaigns: [1],
+        account_info: {
+          first_name: "Test",
+          last_name: "Tests"
+        },
+        setCurrentCampaign: jest.fn()
+      };
+      const wrapper = shallow(<JoinedCampaigns {...campaignsPresentProps} />);
+      expect(wrapper.find('CurrentCampaign').length).toBe(1);
+    });
+    it('should render 0 <CurrentCampaign> when no elements are present in current_campaign prop', () => {
```

Comment:
is the "Join a Campaign" message not shown? I might be misunderstanding this part

Reply:
This component was choking when it wasn't conditionally rendered because it tried to check a property of an object that didn't exist. I think I failed to consider this view when a user has no joined_campaigns because the seed file always includes one. I'll see what I can do.

Reply:
Ohhh. Hmm, if it gets too much maybe we can explore this later when we focus on more UI/UX flows? 👍